runtime/interpreter/interpreter.go

```diff
@@ -3634,6 +3634,45 @@ func (interpreter *Interpreter) IsSubTypeOfSemaType(subType StaticType, superTyp
 	return sema.IsSubType(semaType, superType)
 }
 
+func (interpreter *Interpreter) domainIterator(address common.Address, domain common.PathDomain) []Value {
+	storageMap := interpreter.Storage.GetStorageMap(address, domain.Identifier(), false)
+	if storageMap == nil {
+		return []Value{}
+	}
+	iterator := storageMap.Iterator(interpreter)
+	values := make([]Value, 0, iterator.storage.Count())
+	for key := iterator.NextKey(); key != ""; key = iterator.NextKey() {
+		values = append(values, NewPathValue(interpreter, domain, key))
+	}
+	return values
+}
+
+func (interpreter *Interpreter) publicAccountPaths(addressValue AddressValue, getLocationRange func() LocationRange) *ArrayValue {
+	address := addressValue.ToAddress()
+	values := interpreter.domainIterator(address, common.PathDomainPublic)
+	return NewArrayValue(
+		interpreter,
+		getLocationRange,
+		NewVariableSizedStaticType(interpreter, PrimitiveStaticTypePublicPath),
+		address,
+		values...,
+	)
```

Comment:
One possible solution is to use a temp address for array value, like the approach used by [`interpreterRuntime.newAccountContractsGetNamesFunction()`](https://github.com/onflow/cadence/blob/master/runtime/runtime.go#L2820-L2872).

It calls `NewArrayValue` with `common.Address{}` (temp address).  Because slabs with temp address are not persisted, it avoids the problem of creating unreachable register.  

```suggestion
	return NewArrayValue(
		interpreter,
		getLocationRange,
		NewVariableSizedStaticType(interpreter, PrimitiveStaticTypePublicPath),
		common.Address{},
		values...,
	)
```